app/models/permitted_params.rb

```diff
@@ -33,7 +33,9 @@ def args
       when 'sort_order' then [key, value] if %w(asc desc).include?(value)
       end
     end
-    args.compact.inject({}) { |h, e| h[e.first.to_sym] = e.last; h }
```

Comment:
I'm not sure I understand your question, so let me try to answer:  This particular line was and remains important - if you look the new line 36 you will see that I keep the logic, I just reassign the value to `args` so I can manipulate it further before returning it.  Does that answer your question?